test-storybooks/portable-stories-kitchen-sink/svelte/vite.config.ts

```diff
@@ -1,11 +1,15 @@
-import { defineConfig } from 'vite'
-import { svelte } from '@sveltejs/vite-plugin-svelte'
+import { defineConfig } from 'vite';
+import { svelte } from '@sveltejs/vite-plugin-svelte';
 
 // https://vitejs.dev/config/
-export default defineConfig({
+//@ts-ignore
+export default defineConfig(({ mode }) => ({
   plugins: [svelte()],
   //@ts-ignore
   test: {
     environment: 'jsdom'
-  }
-})
+  },
+  resolve: {
+    conditions: mode === 'test' ? ['browser'] : [],
```

Comment:
@JReinhold and @yannbf I'd need help figuring out why this is happening:
```
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/10]⎯

 FAIL  |@storybook/svelte| src/__test__/composeStories/portable-stories.test.ts > CSF3 > renders with play function with canvas element
Error: warn: %c[svelte] lifecycle_double_unmount
%cTried to unmount a component that was not mounted
 ❯ throwMessage ../../vitest-setup.ts:24:17
     22| const throwMessage = (type: any, message: any) => {
     23|   // eslint-disable-next-line local-rules/no-uncategorized-errors
     24|   const error = new Error(`${type}${message}`);
       |                 ^
     25|   if (!ignoreList.reduce((acc, item) => acc || item(error), false)) {
     26|     throw error;
 ❯ Console.throwWarning ../../vitest-setup.ts:29:40
 ❯ Module.lifecycle_double_unmount ../../node_modules/svelte/src/internal/client/warnings.js:44:11
 ❯ Module.unmount ../../node_modules/svelte/src/internal/client/render.js:292:25
 ❯ unmount ../../node_modules/@testing-library/svelte/src/pure.js:82:64
 ❯ playStory ../../core/dist/preview-api/index.js:5690:42
 ❯ Object.newPlay [as play] ../../core/dist/preview-api/index.js:5618:33
 ❯ src/__test__/composeStories/portable-stories.test.ts:135:32
 ```
 https://app.circleci.com/pipelines/github/storybookjs/storybook/79604/workflows/88023717-27ef-4e54-b730-77feb285891c/jobs/677743

Reply:
I found out that this is caused by the cleanup functions called internally by the new play function mechanism, when you have two tests of the same composed story, where they call play:

```ts
// Error: warn: %c[svelte] lifecycle_double_unmount %cTried to unmount a component that was not mounted

  it.only('renders with custom render function', async () => {
    const Primary = composeStory(stories.CSF3Button, stories.default);

    await Primary.play();
  });

  it.only('renders with custom render function 2', async () => {
    const Primary = composeStory(stories.CSF3Button, stories.default);

    await Primary.play();
  });
  ```
  
  @kasperpeulen could you check it out? The issue is triggered when the cleanup is executed.
  
  Also, FYI this is a warning, however in our current vitest config we make tests break if warnings are logged.

Reply:
I could add that warning to the list of ignored warnings?